components/inputs/input-number.js

```diff
@@ -313,6 +350,56 @@ class InputNumber extends SkeletonMixin(FormElementMixin(LocalizeCoreElement(Lit
 
 	}
 
+	_handleInput(e) {
+		if (e.inputType === 'insertFromPaste') {
+			this._handleChange(e);
+		}
+		if (e.inputType !== 'insertText') {
+			this._hintType = HINT_TYPES.NONE;
+		}
+	}
+
+	_handleKeyPress(e) {
+		const key = e.key;
+		let prevent = false;
+		let hintType = HINT_TYPES.NONE;
+		const hasDecimal = e.target.value.indexOf(this._descriptor.symbols.decimal) > -1;
+		if (key === this._descriptor.symbols.negative) {
+			// negative symbol must be at the end
+			if (this._descriptor.patterns.decimal.negativePattern === '{number}-') {
+				if (e.target.selectionStart !== e.target.value.length) {
+					prevent = true;
+				}
+			} else {
+				// negative symbol must be at the start
+				if (e.target.selectionStart !== 0) {
+					prevent = true;
+				}
+			}
+		} else if (key === this._descriptor.symbols.decimal) {
+			// already has a decimal symbol
+			if (hasDecimal) {
+				hintType = HINT_TYPES.DECIMAL_DUPLICATE;
+				prevent = true;
+			// looking for an integer only
+			} else if (this.maxFractionDigits === 0) {
+				hintType = HINT_TYPES.INTEGER;
+				prevent = true;
+			}
+		} else if ((key === '.' || key === ',') && !hasDecimal && this.maxFractionDigits > 0) {
+			// incorrect decimal
+			hintType = key === ',' ? HINT_TYPES.DECIMAL_INCORRECT_PERIOD : HINT_TYPES.DECIMAL_INCORRECT_COMMA;
+			prevent = true;
+		} else if (['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'].indexOf(key) === -1) {
```

Comment:
Is `e` a valid character?  The reason I ask is - the native `input type="number"` appears to suppress input, but it allows the `e` character.  I'm guessing we can't use the native type="number" due to our crazy localization rules, ya?  The native number input includes up/down arrows, which undoubtedly would throw a little wrench into our nice layout. Nevertheless, mobile devices may provide a nicer keyboard (showing numbers only) for native number inputs.  But, there may be some browser inconsistency with native number input also, as described [in this issue](https://bugzilla.mozilla.org/show_bug.cgi?id=1398528).

Reply:
Yeah native number input isn't in the cards for us right now for the reasons you listed. It's totally inconsistently implemented across browsers and more importantly it doesn't support different decimal separators. Had we never supported those we might be able to get away with it, but changing that behaviour at this point would be very difficult.

Reply:
Sounds good.  Do we need to allow the `e` character though?

Reply:
We've never allowed scientific notation in our numeric inputs -- the "e" character results in an "Invalid Decimal" error with our old version -- so I don't think we need to worry about it at this point.